docs/New Game.md

```diff
@@ -31,14 +31,82 @@ game to have proper pickups:
+In most case, this involves readings the files in
+a user-provided copy of the game and creating either a new ROM/ISO file or a mod file.
 
-**TODO**: details
+This process is split into two parts:
+
+### Patch Data Factory
+
+Each game defines a class that inherits from `PatchDataFactory`. Given a LayoutDescription, a CosmeticPatches and 
+PlayersConfiguration, it creates a JSON Object that will be passed as an argument to the Game Exporter.
+
+### Game Exporter
+
+Each game also defines a class that inherits from `GameExporter`. Given the JSON created by the PatchDataFactory and a
+GameExportParams, provided by your game's GameExportDialog, it's expected to create all the necessary files for the
+user to play the game.
+
+Since modifying the game files requires nothing from Randovania and can be a quite big task, a separate python package 
+responsible for this task is created. This is called the patcher for the game. You are free to implement this package
+however you want, but we recommend the following:
+
+#### Input Data
+
+The arguments for the patcher should be:
+- A Json file, called the patcher data.
+- Paths to needed game files
+- Paths to where to output goes.
+- A status update function
+
+Nothing more. Any other parameter should come from the json file.
+
+#### Language
+
+Best:
+- Python
+
+Good (Known to integrate without issues):
+- C
+- C++
+- Rust
+- Lua
+
+Likely Good (Should integrate well, never tested):
+- Languages that compile to static library, such a Go
+
+Ok (Expect additional work to integrate properly, avoid if viable):
+- C#
+
+Avoid languages that require the user to install a runtime, such as Java.
+
+#### Standalone Use
+
+Allow your patcher to be used directly in the command line, by passing the game files and a json file.
+
+This makes it easy to test the patcher without having to do it through Randovania, and helps identify issues better.
+
+#### Json Schema
+
+Define a [JSON Schema](https://json-schema.org/) that defines the structure of the patcher data. This makes for a good
+documentation of what is the input for the patcher, as well as a very good safety against misconfiguration in Randovania. 
 
 
 ### Multiworld
 
 **TODO**
+
+# Rules for new games
+
+Randovania has a high bar of quality for our game integrations, and we wish to keep it that way. In order to help
+with that goal, there are rules that need to be met before we can accept a game.
+
+- An owner for the game, with an expectation to be available for a while.
+- A reviewer of Logic changes, separate from the owner.
+- An open source patcher, with a license similar to GPL or Apache.
+- Proper test coverage of new code.
```

Comment:
A) Are these *rules* or *guidelines*?

B) any further tips on how to do test coverage? this is a bit tricky, since it usually requires copyrighted files.-

Reply:
A) Intentionally ambiguous. But my intention is that everything in this last section is a requirement, but we reserve the right to make exceptions.

B) There's plenty of code one adds to Randovania that doesn't need game data. But I'm trying to find a reliable way to do tests with access to game data.

Reply:
>  Intentionally ambiguous

Currently from reading it, it very much feels like "these are set in stone rules without exceptions", hence why me asking